pinot-core/src/test/java/org/apache/pinot/core/data/manager/realtime/RealtimeConsumptionRateManagerTest.java

```diff
@@ -116,4 +121,40 @@ public void testBuildCache() throws Exception {
     assertEquals((int) cache.get(STREAM_CONFIG_C), 1); // use cache
     verify(partitionCountFetcher, times(1)).fetch(STREAM_CONFIG_C);
   }
+
+  @Test
+  public void testMetricEmitter() {
+    // setup metric emitter
+    double rateLimit = 2; // 2 msgs/sec = 120 msgs/min
+    ServerMetrics serverMetrics = mock(ServerMetrics.class);
+    MetricEmitter metricEmitter = new MetricEmitter(serverMetrics, "tableA-topicB-partition5");
+
+    // 1st minute: no metrics should be emitted in the first minute
+    Instant now = Clock.fixed(Instant.parse("2022-08-10T12:00:02Z"), ZoneOffset.UTC).instant();
+    assertEquals(metricEmitter.emitMetric(10, rateLimit, now), 0);
+    now = Clock.fixed(Instant.parse("2022-08-10T12:00:10Z"), ZoneOffset.UTC).instant();
+    assertEquals(metricEmitter.emitMetric(20, rateLimit, now), 0);
+    now = Clock.fixed(Instant.parse("2022-08-10T12:00:30Z"), ZoneOffset.UTC).instant();
+    assertEquals(metricEmitter.emitMetric(5, rateLimit, now), 0);
+    now = Clock.fixed(Instant.parse("2022-08-10T12:00:55Z"), ZoneOffset.UTC).instant();
+    assertEquals(metricEmitter.emitMetric(25, rateLimit, now), 0);
+
+    // 2nd minute: metric should be emitted
+    now = Clock.fixed(Instant.parse("2022-08-10T12:01:05Z"), ZoneOffset.UTC).instant();
+    assertEquals(metricEmitter.emitMetric(35, rateLimit, now), (int) Math.round((10 + 20 + 5 + 25) / 120.0 * 100));
```

Comment:
can we have a comment to explain this calculation?

Reply:
I would suggest to use constants and name them startTime, startTimeplusFiveSec etc, that will make code very easy to read, this also avoids the need of comments like 3rd minute, 4th minute etc

Reply:
I thought having the numbers there shows how the calculation is done, but apparently it wasn't clear enough! Did refactor to better explain the numbers, however I kept the separation of what happens in each minute so it's more clear what we want to achieve there.